### PR TITLE
Use Chef DK for testing

### PR DESCRIPTION
**Pull request by shoekstra**
* Switch bundler/gems to Chef DK as it already has all tools needed for testing
* Switch rubocop to cookstyle
* Add a .rubocop_todo.yml for cookstyle warnings
* Add a rake task to test style and spec
* Deleted Gemfile

Signed-off-by: Stephen Hoekstra <shoekstra@schubergphilis.com>

### CODE REVIEW

**Comment by shoekstra** [comment]
Any news on this PR?  I have more to add but would like working CI tests before continuing (and fixing that seemed a good place to start)... 

/cc @josephholsten @JCotton1123 @mjuszczak @jwmarshall @sinfomicien @joerocklin (found your names [here](https://github.com/sinfomicien/mariadb/issues/100))

**Comment by mjuszczak** [comment]
I have not had a chance to review, sorry, but it’s on my list.

File: Gemfile [hunk 1]
```diff
@@ -1,46 +0,0 @@
-source 'https://rubygems.org'
```
Comment (josephholsten) [inline Gemfile:1]: Any particular reason you're removing the gemfile? I there's a decent number of things that aren't in chefdk listed here.

**Review by josephholsten — APPROVED** [review]
This is fine, but we might want to keep the gemfile.

File: Gemfile (hunk 1 above)
Comment (shoekstra) [reply Gemfile:1]: Everything needed is baked into ChefDK - if the need arises I will create a new Gemfile (e.g. for deploying to supermarket with the stove gem), but until now it's just fluff.

**Pull request merged**